### ISSUE TITLE
Empty dictionary returned from grains.items command within local client

### ISSUE THREAD

**Issue by UtahCampusD** [saltstack/salt#35813]
### Description of Issue/Question
The grains.items command returns an empty dictionary when calling it from within the local client (salt.client)

### Setup & Steps to Reproduce Issue
import salt.client
caller = salt.client.Caller()
grains_list = caller.cmd('grains.items')
print grains_list
{}

### Versions Report
$ salt-minion --versions-report
Salt Version:
           Salt: 2016.3.2
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.20.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, Aug 18 2016, 15:13:37)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: centos 6.8 Final
        machine: x86_64
        release: 2.6.32-642.3.1.el6.centos.plus.x86_64
         system: Linux
        version: CentOS 6.8 Final

**Comment by Ch3LL** [comment]
@UtahCampusD looks like I am able to replicate this as shown below:

```
>>> caller.cmd('grains.item', 'os')
{'os': 'Fedora'}
>>> caller.cmd('grains.items')
{}
```

Also to note it seems `pillar.items` does work

Looks like we need to get this fixed up. Thanks

**Comment by cachedout** [comment]
Fixed in #36496. Please give it a spin.
